components/filemanager/toolbar.md

```diff
@@ -10,36 +10,403 @@ position: 15
+## Custom Toolbar
 
+To customize the Toolbar, add the `<FileManagerToolBar>` tag as a direct child of the `<TelerikFileManager>`. Inside the `<FileManagerToolBar>` you can include your desired [built-in tools](#built-in-tools), manage their order and add [custom tools](#custom-tools).
 
-## Views
+### Built-in tools
 
-The `Views` button group allows you to control how the content of the selected directory will be visualized. The users can choose between two options. Read more in the [FileManager - Views]({%slug filemanager-views%}) article.
+To add the desired built-in tools, declare the corresponding tool components in the `<FileManagerToolBar>`. The tools will be rendered in the order that you declare them in.
 
-## View Details
+<div class="skip-repl"></div>
 
-The `View Details` toggles the visibility of the [Preview Pane]({%slug filemanager-preview-pane%}). By default is is `Off` and you can enable it to see the selected file details.
+````CSHTML
+<TelerikFileManager Data="@FileManagerData">
+                   <FileManagerToolBar>
+                       <FileManagerToolBarSearchTool/>
+                   </FileManagerToolBar>
+</TelerikFileManager>
+````
 
-## Search
+>caption Built-in tool components:
 
-The FileManager allows searching files in the current location via a Searchbar in the Toolbar. Read more in the [FileManager - Search]({%slug filemanager-search%}) article.
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
+
+| Tool Component | Description |
+|------|-------------|
+| `FileManagerToolBarNewFolderTool` | Renders the button for creating a new folder. |
+| `FileManagerToolBarUploadTool` | Renders the button for the upload window. |
+| `FileManagerToolBarSortDirectionTool` | Renders a ButtonGroup for the sort direction. |
+| `FileManagerToolBarSortTool` | Renders the SplitButton for choosing which field to sort by. |
+| `FileManagerToolBarFileViewTool` | Renders a ButtonGroup to toggle between Grid and ListView views. |
+| `FileManagerToolBarSpacer` | Renders a spacer element. Use that if you need to add some space between the declared tools. |
+| `FileManagerToolBarViewDetailsTool` | Renders a Switch to toggle the Preview Pane for the selected file. |
+| `FileManagerToolBarSearchTool` | Renders a search box. | 
+
+### Custom tools
+
+To create a custom tool, declare a `<FileManagerToolBarCustomTool>` inside `<FileManagerToolBar>` tag. Place the content of your tool inside the `<FileManagerToolBarCustomTool>` tag.
+
+<div class="skip-repl"></div>
+
+````CSHTML
+<TelerikFileManager Data="@FileManagerData">
+                   <FileManagerToolBar>
+                       <FileManagerToolBarCustomTool>
+                           <TelerikButton>Custom tool</TelerikButton>
+                       </FileManagerToolBarCustomTool>
+                   </FileManagerToolBar>
+</TelerikFileManager>
+````
+
+### Example
```

Comment:
Using JavaScript for a built-in feature is a but unusual. Can we think of something simpler for our customers? For example:
* The custom tool can perform an action that is normally in the context menu (delete).
* The custom tool can display the value of a custom property in the data item.